File: push/src/main/java/com/xiaomi/push/service/XMPushServiceReceivers.java
package com.xiaomi.push.service;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

final class ConnectionChangeReceiver extends BroadcastReceiver {
    private final XMPushService service;

    ConnectionChangeReceiver(XMPushService xMPushService) {
        this.service = xMPushService;
    }

    @Override
    public void onReceive(Context context, Intent intent) {
        this.service.onStart(intent, 1);
    }
}

final class ScreenStateReceiver extends BroadcastReceiver {
    private final XMPushService service;

    ScreenStateReceiver(XMPushService xMPushService) {
        this.service = xMPushService;
    }

    @Override
    public void onReceive(Context context, Intent intent) {
        this.service.onStart(intent, 1);
    }
}

File: legacy/src/main/java/com/xiaomi/push/service/XMPushServiceReceivers.kt
package com.xiaomi.push.service
import io.github.magisk317.mipush.protocol.model.*

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

class ConnectionChangeReceiver(
    private val service: XMPushService,
) : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        service.onStartCommand(intent, 0, 1)
    }
}

class ScreenStateReceiver(
    private val service: XMPushService,
) : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        service.onStartCommand(intent, 0, 1)
    }
}
